Updating Sauce Labs to include saucectl (#5529)
circleci/circleci-docs: jekyll/_cci2/browser-testing.md

## Sauce Labs
{: #sauce-labs }

> Sauce Labs has an extensive network of operating system and browser combinations you can test your web application against. Sauce Labs supports automated web app testing using Selenium WebDriver scripts as well as through `saucectl` - their test orchestrator CLI that can execute tests directly from a variety of JavaScript frameworks.

Reviewer: ```suggestion
Sauce Labs has an extensive network of operating system and browser combinations you can test your web application against. Sauce Labs supports automated web app testing using Selenium WebDriver scripts as well as through `saucectl`, their test orchestrator CLI, which can be used to execute tests directly from a variety of JavaScript frameworks.
```

### saucectl
{: #saucectl }

> If you're using JavaScript to test your web application, you can still take advantage of the Sauce Labs platform by using [`saucectl`](https://docs.saucelabs.com/testrunner-toolkit) with the JS framework of your choice, and then integrating the [saucectl-run Orb](https://circleci.com/developer/orbs/orb/saucelabs/saucectl-run) in your CircleCI workflow.

Reviewer: ```suggestion
If you are using JavaScript to test your web application, you can still take advantage of the Sauce Labs platform by using [`saucectl`](https://docs.saucelabs.com/testrunner-toolkit) with the JS framework of your choice, and then integrating the [saucectl-run orb](https://circleci.com/developer/orbs/orb/saucelabs/saucectl-run) in your CircleCI workflow.
```

> 1. Add your `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` as Environment variables in your Circle CI project.

Reviewer: ```suggestion
1. Add your `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` as [environment variables]({{site.baseurl}}/2.0/env-vars/) in your Circle CI project.
```

2. Modify your CircleCI project `config.yml` to include the saucectl-run orb and then call the orb as a job in your workflow.

{% raw %}
```yaml
version: 2.1
orbs:

>   saucectl: saucelabs/saucectl-run@1.0.0

Reviewer 2: Update to 1.1.0 when this orb update goes live: https://github.com/saucelabs/saucectl-run-orb/pull/3#pullrequestreview-770322950

Author: Thanks, @KyleTryon. Any estimates on when that will happen?

Author: Okay. I have tee'd up the change in anticipation of the orb update.

Reviewer 3: @sweeneyskirt-sl I'll ping you when it's available !
 It will be version 2.0.0 as we introduced a breaking change.

>   saucectl: saucelabs/saucectl-run@1.0.0

Author: ```suggestion
  saucectl: saucelabs/saucectl-run@1.1.0
```

>   saucectl: saucelabs/saucectl-run@1.1.0

Reviewer 3: ```suggestion
  saucectl: saucelabs/saucectl-run@2.0.0
```

```
{% endraw %}

> For more detailed information about using the Sauce Connect orb in your workflows, refer to the [Sauce Connect Orb](https://circleci.com/developer/orbs/orb/saucelabs/sauce-connect) page in the [CircleCI Orbs Registry](https://circleci.com/developer/orbs).

Reviewer 2: @KunalJain It looks like we still need to have this orb updated if there is anyone you can put us in contact with, I know we started this once but maybe ran into issues.
https://circleci.com/developer/orbs/orb/saucelabs/sauce-connect

Author: I am not sure which team owns the sauce-connect orb, but I will see if I can ferret it out. Thanks!

Author: Hi @KyleTryon -- sorry for the delay on this. I have spoken with a number of folks at Sauce Labs, and it seems the Sauce Connect Orb development team has disbanded and there are no plans at this time to devote resources to it. I would like to propose an EOL to the Sauce Connect orb in its current state and amend this doc update to remove the Sauce Connect section so we might move forward with the update related to saucectl. Does that sound reasonable?
cc: @KunalJain

### saucectl
{: #saucectl }

If you're using JavaScript to test your web application, you can still take advantage of the Sauce Labs platform by using [`saucectl`](https://docs.saucelabs.com/testrunner-toolkit) with the JS framework of your choice, and then integrating the [saucectl-run Orb](https://circleci.com/developer/orbs/orb/saucelabs/saucectl-run) in your CircleCI workflow.

1. Add your `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` as Environment variables in your Circle CI project.
2. Modify your CircleCI project `config.yml` to include the saucectl-run orb and then call the orb as a job in your workflow.

{% raw %}
```yaml
version: 2.1
orbs:
  saucectl: saucelabs/saucectl-run@1.0.0

jobs:
  test-cypress:
    docker:

>       - image: circleci/node:12.21

Reviewer 2: Recommending updating docker images to use our next-gen images: https://circleci.com/developer/images/image/cimg/node

An aside, it may be difficult to keep this example in-sync with orb updates. Orbs themselves hold usage examples as well. I think it might be worth considering linking to an example in the orb rather than including it here.

Author: Thanks for the feedback, @KyleTryon. I am going to circle back with the dev team to see if the orb has an example we would prefer to reference here and make sure the docker image is current.

Reviewer 3: @KyleTryon  is there any update to be done to the orb itself, or that's just this example and our examples that needs to be updated ?

Reviewer: @KyleTryon Any more feedback here? I think it's just the image used that needs updating but wanted to check in with you to confirm :-)

Reviewer: Hey, could we just update to use a next gen image for this example then we can get these changes merged, and as you say move the wider orb update conversation outside this PR?

Author: Thanks @rosieyohannan! I have made the update to the image reference. Our developer has also tested the orb using the latest image to verify it works and has made the changes to our examples in the circleCI repo.
